### PR TITLE
System info screen

### PR DESCRIPTION
**Pull request by algernon**
This adds a new screen to Chrysalis, which allows one to let Chrysalis collect information about the system (OS, Chrysalis itself, and any connected device) and bundle that up in a zip file they can send us.

Screenshot from 2020-10-08 14-22-15

Clicking the "Collect Information" button will gather the info, and then pop up a save dialog:

Screenshot from 2020-10-08 14-22-37

At the moment, the following files are included in the zip file:

```
❯ unzip -l chrysalis-debug.bundle.zip
Archive:  chrysalis-debug.bundle.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      321  2020-10-08 12:23   osInfo.json
       20  2020-10-08 12:23   chrysalis.json
    34849  2020-10-08 12:23   logs.json
    24319  2020-10-08 12:23   device.json
---------                     -------
    59509                     4 files
```

## `osInfo.json`

```json
{
  "platform": "linux",
  "distro": "Debian GNU/Linux",
  "release": "10",
  "codename": "buster",
  "kernel": "4.19.0-11-amd64",
  "arch": "x64",
  "hostname": "fangorn",
  "codepage": "UTF-8",
  "logofile": "debian",
  "serial": "2a753bbe2f99452893073d05bfd9cbb1",
  "build": "",
  "servicepack": "",
  "uefi": false
}
```

## `chrysalis.json`

```json
{"version": "0.7.9"}
```

`logs.json` contains all our logs, the same ones we send to the developer console, while `device.json` contains information about the device:

```json
{
  "info": {
    "vendor": "Keyboardio",
    "product": "Model01",
    "displayName": "Keyboardio Model01",
    "urls": [
      {"name": "Homepage", "url": "https://shop.keyboard.io/"},
      {"name": "Forum", "url": "https://community.keyboard.io/"},
      {"name": "Chat", "url": "https://keyboard.io/discord-invite"}
    ]
  },
  "path": "/dev/ttyACM0",
  "commands": [
    "help",
    "version",
    "keymap.custom",
    "keymap.default",
    "keymap.onlyCustom",
    "palette",
    "colormap.map",
    "settings.defaultLayer",
    "settings.valid?",
    "settings.version",
    "settings.crc",
    "eeprom.contents",
    "eeprom.free",
    "hostos.type",
    "cycletime"
  ],
  "keymap": {
    "onlyCustom": true,
    "custom": [...],
    "default": [...]
  },
  "colormap": {
    "palette": [...],
    "colorMap": [...]
  }
}
```

### CODE REVIEW

File: src/renderer/i18n/en.js [hunk 1]
```diff
@@ -211,6 +212,16 @@ const English = {
     gotoUpdate: "Update Firmware",
     reconnect: "Reconnect",
     reconnectDescription: `There's a possibility that we misdetected the capabilities of the keyboard, or that the keyboard was starting up while we connected. In this case, you can try clicking the "{{buttonName}}" button to attempt a reconnect, and look for the necessary features again. Reconnecting is useful if you're sure there was a temporary failure upon previous attempts, and the problem has been resolved.`
+  },
+  systeminfo: {
+    title: "System Information",
+    contents: `Chrysalis can collect a debug bundle you can share with the developers as a troubleshooting aid. The bundle will include information about your operating system and its version; Chrysalis' version; and if there's a connected device, its name, and selected properties of it (such as the keymap and the colormap).`,
```
Comment (obra) [inline src/renderer/i18n/en.js:218]: If you're having trouble with Chrysalis, the development team may ask you to send them some debugging information. Chrysalis can help you collect and package up everything you might need to share. This includes error logs, as well as a bit of information about this computer's operating system, connected devices, and Chrysalis itself. Keyboard information will include information about the keyboard's current firmware, as well as key layouts and LED configuration. 

We try hard not to include private information in this bundle, but recommend you read through it before sharing it.

File: src/renderer/i18n/en.js [hunk 2]
```diff
@@ -211,6 +212,16 @@ const English = {
...
+    collectButton: "Collect information",
```
Comment (obra) [inline src/renderer/i18n/en.js:219]: Create Bundle

**Review by obra — CHANGES_REQUESTED** [review]

**Comment by obra** [comment]
Awesome.

How hard would it be to show all of the information in a textarea on the save bundle page?

**Comment by algernon** [comment]
It wouldn't be hard, but it's a _lot_ of info. How about I turn the `Create bundle` button into `View bundle` when the bundle is done? That'd pop up a dialog with a textarea, and a `Save bundle` button on top. (And we wouldn't have a save dialog appear during create then)

**Comment by obra** [comment]
Thinking a little bit about the format - I think we either just want one json file or we want all of the files to include fingerprint information to tie them together:

reportInfo: { 
   chrysalisVersion: Blah,
   timestamp: DATETIMETHATISHTESAMEFORALLFILES,
   reportId: UUID-THAT-IS-THE-SAME-FOR-ALL-FILES-IN-THIS-REPORT
}

**Comment by algernon** [comment]
I'd go with a single file then, that's a bit easier. But a timestamp and an UUID is a good idea regardless, so I'll add those to that single file anyway.

**Comment by algernon** [comment]
Oh, and I'll still zip it, to make the bundle format future proof, in case we want to include screenshots or whatever later on.

**Comment by algernon** [comment]
Updates:

## System Information

Screenshot from 2020-10-09 02-58-42

## After creating the bundle...

Screenshot from 2020-10-09 02-58-45

## View Bundle

Screenshot from 2020-10-09 02-58-52

## After saving

Screenshot from 2020-10-09 02-59-02

# Notes

When the `View Bundle` dialog is closed in any way, either by saving the bundle, or clicking the button, or pressing `Esc`, the component resets, and the user will have to create another bundle. The bundle now contains a single `bundle.json` file, which includes a timestamp and an UUID. The UUID isn't all that useful yet, but if we will include other files in the future, it will make it easier to match files up.

I'm pretty happy with the current state, so marking this as ready.

**Pull request merged**